tempodb/encoding/vparquet5/block_search.go

```diff
@@ -0,0 +1,487 @@
+		if column == "" {
+			otherAttrConditions[k] = v
+			continue
+		}
+
+		// most columns are just a substring predicate over the column, but we have
+		// special handling for http status code and span status
+		if k == LabelHTTPStatusCode {
+			if i, err := strconv.Atoi(v); err == nil {
+				resourceIters = append(resourceIters, makeIter(column, pq.NewIntBetweenPredicate(int64(i), int64(i)), ""))
+				continue
+			}
+			// Non-numeric string field
+			otherAttrConditions[k] = v
+			continue
+		}
+		if k == LabelStatusCode {
+			code := StatusCodeMapping[v]
+			resourceIters = append(resourceIters, makeIter(column, pq.NewIntBetweenPredicate(int64(code), int64(code)), ""))
+			continue
+		}
+
+		if k == LabelRootServiceName || k == LabelRootSpanName {
+			traceIters = append(traceIters, makeIter(column, pq.NewSubstringPredicate(v), ""))
+		} else {
+			resourceIters = append(resourceIters, makeIter(column, pq.NewSubstringPredicate(v), ""))
+		}
+	}
+
+	// Generic attribute conditions?
+	if len(otherAttrConditions) > 0 {
+		// We are looking for one or more foo=bar attributes that aren't
+		// projected to their own columns, they are in the generic Key/Value
+		// columns at the resource or span levels.  We want to search
+		// both locations. But we also only want to read the columns once.
+
+		keys := make([]string, 0, len(otherAttrConditions))
+		vals := make([]string, 0, len(otherAttrConditions))
+		for k, v := range otherAttrConditions {
+			keys = append(keys, k)
+			vals = append(vals, v)
+		}
+
+		keyPred := pq.NewStringInPredicate(keys)
+		valPred := pq.NewStringInPredicate(vals)
+
+		// This iterator combines the results from the resource
+		// and span searches, and checks if all conditions were satisfied
+		// on each ResourceSpans.  This is a single-pass over the attribute columns.
+		j := pq.NewUnionIterator(DefinitionLevelResourceSpans, []pq.Iterator{
+			// This iterator finds all keys/values at the resource level
+			pq.NewJoinIterator(DefinitionLevelResourceAttrs, []pq.Iterator{
+				makeIter(FieldResourceAttrKey, keyPred, "keys"),
+				makeIter(FieldResourceAttrVal, valPred, "values"),
+			}, nil),
+			// This iterator finds all keys/values at the span level
+			pq.NewJoinIterator(DefinitionLevelResourceSpansILSSpanAttrs, []pq.Iterator{
+				makeIter(FieldSpanAttrKey, keyPred, "keys"),
+				makeIter(FieldSpanAttrVal, valPred, "values"),
+			}, nil),
+		}, pq.NewKeyValueGroupPredicate(keys, vals))
+
+		resourceIters = append(resourceIters, j)
+	}
+
+	// Multiple resource-level filters get joined and wrapped
+	// up to trace-level. A single filter can be used as-is
+	if len(resourceIters) == 1 {
+		traceIters = append(traceIters, resourceIters[0])
+	}
+	if len(resourceIters) > 1 {
+		traceIters = append(traceIters, pq.NewJoinIterator(DefinitionLevelTrace, resourceIters, nil))
+	}
+
+	// Duration filtering?
+	if req.MinDurationMs > 0 || req.MaxDurationMs > 0 {
+		mn := int64(0)
```

Comment:
`nitpick` Still not finding `mn` and `mx` very intuitive names. Maybe `lower`/`upper` or `start`/`end`?

Reply:
Updated